dali/python/nvidia/dali/pool.py

```diff
@@ -0,0 +1,265 @@
+# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import socket
+import pickle
+import threading
+import multiprocessing
+from collections import OrderedDict
+from nvidia.dali import backend as _b
+from nvidia.dali.worker import worker
+from nvidia.dali.shared_batch import deserialize_batch, import_numpy
+from nvidia.dali import shared_mem
+
+
+class MemChunk(object):
+
+    def __init__(self, shm_chunk, version, capacity):
+        self.shm_chunk = shm_chunk
+        self.version = version
+        self.capacity = capacity
+
+
+class SharedBatchesConsumer(object):
```

Comment:
Hmm, maybe you have an idea for a better naming here? Because the main functionality of this class is keeping track of open shared memory chunks, so it is more akin to the BatchDispatcher class (I change its name to SharedBatchesDispatcher for uniformity) that is defined in the worker.py.